service/common/src/main/java/org/apache/polaris/service/admin/PolarisAdminService.java

```diff
@@ -699,16 +701,10 @@ private Map<String, UserSecretReference> extractSecretReferences(
   /**
    * @see #extractSecretReferences
    */
-  private boolean requiresSecretReferenceExtraction(CreateCatalogRequest catalogRequest) {
-    Catalog catalog = catalogRequest.getCatalog();
-    if (catalog instanceof ExternalCatalog externalCatalog) {
-      if (externalCatalog.getConnectionConfigInfo() != null) {
-        // TODO: Make this more targeted once we have connection configs that don't involve
-        // processing of inline secrets.
-        return true;
-      }
-    }
-    return false;
+  private boolean requiresSecretReferenceExtraction(
+      @NotNull ConnectionConfigInfo connectionConfigInfo) {
+    return connectionConfigInfo.getAuthenticationParameters().getAuthenticationType()
+        != AuthenticationParameters.AuthenticationTypeEnum.IMPLICIT;
```

Comment:
This will return `true` for `NULL_TYPE`... Is that intended?

Reply:
The type here refers to `AuthenticationTypeEnum` (generated based on the OpenAPI definition) and not `AuthenticationType` (declared in `...core.connection.AuthenticationType`). So `AuthenticationTypeEnum` can never be `NULL_TYPE`, it will be one of the types defined in polaris-management-service.yml (Line 910 - 914).

Reply:
Ah, sorry, I misinterpreted :facepalm: